From 82bf02188ce61c532e99a81972c549bf00748928 Mon Sep 17 00:00:00 2001
From: auto commit <unknown@unknown>
Date: Fri, 24 Oct 2008 12:22:46 -0700
Subject: [PATCH] cavium: Add in CAVIUM_RESERVE32_USE_WIRED_TLB

Add in the task size changes associated with the
CAVIUM_RESERVE32_USE_WIRED_TLB optimization.

Signed-off-by: Tomaso Paoletti <tpaoletti@caviumnetworks.com>
Signed-off-by: Paul Gortmaker <Paul.Gortmaker@windriver.com>
---
 arch/mips/kernel/binfmt_elfn32.c |    4 ++++
 arch/mips/kernel/binfmt_elfo32.c |    4 ++++
 include/asm-mips/processor.h     |    4 ++++
 3 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/arch/mips/kernel/binfmt_elfn32.c b/arch/mips/kernel/binfmt_elfn32.c
index 9fdd8bc..067985d 100644
--- a/arch/mips/kernel/binfmt_elfn32.c
+++ b/arch/mips/kernel/binfmt_elfn32.c
@@ -46,7 +46,11 @@ typedef elf_fpreg_t elf_fpregset_t[ELF_NFPREG];
 	__res;								\
 })
 
+#ifdef CONFIG_CAVIUM_RESERVE32_USE_WIRED_TLB
+#define TASK32_SIZE		(0x7fff8000UL - (CONFIG_CAVIUM_RESERVE32<<20))
+#else
 #define TASK32_SIZE		0x7fff8000UL
+#endif
 #undef ELF_ET_DYN_BASE
 #define ELF_ET_DYN_BASE         (TASK32_SIZE / 3 * 2)
 
diff --git a/arch/mips/kernel/binfmt_elfo32.c b/arch/mips/kernel/binfmt_elfo32.c
index ab8268a..0e061c2 100644
--- a/arch/mips/kernel/binfmt_elfo32.c
+++ b/arch/mips/kernel/binfmt_elfo32.c
@@ -48,7 +48,11 @@ typedef elf_fpreg_t elf_fpregset_t[ELF_NFPREG];
 	__res;								\
 })
 
+#ifdef CONFIG_CAVIUM_RESERVE32_USE_WIRED_TLB
+#define TASK32_SIZE		(0x7fff8000UL - (CONFIG_CAVIUM_RESERVE32<<20))
+#else
 #define TASK32_SIZE		0x7fff8000UL
+#endif
 #undef ELF_ET_DYN_BASE
 #define ELF_ET_DYN_BASE         (TASK32_SIZE / 3 * 2)
 
diff --git a/include/asm-mips/processor.h b/include/asm-mips/processor.h
index 0840713..7602e4b 100644
--- a/include/asm-mips/processor.h
+++ b/include/asm-mips/processor.h
@@ -56,7 +56,11 @@ extern unsigned int vced_count, vcei_count;
  * support 16TB; the architectural reserve for future expansion is
  * 8192EB ...
  */
+#ifdef CONFIG_CAVIUM_RESERVE32_USE_WIRED_TLB
+#define TASK_SIZE32	(0x7fff8000UL - (CONFIG_CAVIUM_RESERVE32<<20))
+#else
 #define TASK_SIZE32	0x7fff8000UL
+#endif
 #define TASK_SIZE	0x10000000000UL
 #define STACK_TOP	\
       (test_thread_flag(TIF_32BIT_ADDR) ? TASK_SIZE32 : TASK_SIZE)
-- 
1.5.5.1

